e2e/testsuite/tx.go

```diff
@@ -278,9 +277,28 @@ func (s *E2ETestSuite) ExecuteGovV1Beta1Proposal(ctx context.Context, chain ibc.
 
 // Transfer broadcasts a MsgTransfer message.
 func (s *E2ETestSuite) Transfer(ctx context.Context, chain ibc.Chain, user ibc.Wallet,
-	portID, channelID string, token sdk.Coin, sender, receiver string, timeoutHeight clienttypes.Height, timeoutTimestamp uint64, memo string,
+	portID, channelID string, tokens sdk.Coins, sender, receiver string, timeoutHeight clienttypes.Height, timeoutTimestamp uint64, memo string,
 ) sdk.TxResponse {
-	msg := transfertypes.NewMsgTransfer(portID, channelID, sdk.NewCoins(token), sender, receiver, timeoutHeight, timeoutTimestamp, memo)
+	channel, err := query.Channel(ctx, chain, portID, channelID)
+	s.Require().NoError(err)
+	s.Require().NotNil(channel)
+
+	feeEnabled := false
+	if testvalues.FeeMiddlewareFeatureReleases.IsSupported(chain.Config().Images[0].Version) {
+		feeEnabled, err = query.FeeEnabledChannel(ctx, chain, portID, channelID)
+		s.Require().NoError(err)
+	}
+
+	transferVersion := channel.Version
+	if feeEnabled {
+		version, err := feetypes.MetadataFromVersion(channel.Version)
+		s.Require().NoError(err)
+
+		transferVersion = version.AppVersion
+	}
```

Comment:
Query the channel to get the version and to see if it's fee enabled to decode the JSON-encoded version string and get the transfer app version.